sklearn/linear_model/_ridge.py

```diff
@@ -1816,36 +1815,35 @@ def __init__(self, alphas=(0.1, 1.0, 10.0), fit_intercept=True,
         self.class_weight = class_weight
 
     def fit(self, X, y, sample_weight=None):
-        """Fit the ridge classifier.
+        """Fit Ridge classifier with cv.
 
         Parameters
         ----------
-        X : array-like, shape (n_samples, n_features)
+        X : array-like of shape (n_samples, n_features)
             Training vectors, where n_samples is the number of samples
             and n_features is the number of features. When using GCV,
             will be cast to float64 if necessary.
 
-        y : array-like, shape (n_samples,)
+        y : array-like of shape (n_samples,)
             Target values. Will be cast to X's dtype if necessary
 
-        sample_weight : {float, array-like of shape (n_samples,)}, default=None
+        sample_weight : float or array-like of shape (n_samples,), default=None
             Sample weight.
 
         Returns
         -------
         self : object
         """
-        check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'],
-                  multi_output=True)
+        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'coo'],
+                         multi_output=True, y_numeric=False)
```

Comment:
Looks like a bug indeed.

Reply:
Not really, but it is intricate. `RidgeClassifierCV.fit` calls `_BaseRidgeCV.fit`, which calls either `Ridge.fit`or `_RidgeGCV.fit`. The last two also do the usual `X, y = check_X_y`. Nevertheless, I prefer to have it explicitly there.